### PR TITLE
Refactor DatabaseConfig related things

### PR DESCRIPTION
**Pull request by brfrn169**
This PR refactors DatabaseConfig related things. I will add inline comments for the details. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/config/DatabaseConfig.java [hunk 1]
```diff
@@ -89,6 +89,10 @@ public DatabaseConfig(Properties properties) {
     load();
   }
 
+  public DatabaseConfig(Path propertiesPath) throws IOException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/config/DatabaseConfig.java:92]: Added a constructor where we can pass a `Path` object.

File: core/src/main/java/com/scalar/db/service/StorageFactory.java [hunk 2]
```diff
@@ -27,8 +37,32 @@ public DistributedStorage getStorage() {
...
+    return new StorageFactory(new DatabaseConfig(properties));
+  }
+
+  public static StorageFactory create(Path propertiesPath) throws IOException {
+    return new StorageFactory(new DatabaseConfig(propertiesPath));
+  }
+
+  public static StorageFactory create(File propertiesFile) throws IOException {
+    return new StorageFactory(new DatabaseConfig(propertiesFile));
+  }
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/service/StorageFactory.java:67]: Added several factory methods to `StorageFactory` and `TransactionFactory`.

File: core/src/main/java/com/scalar/db/storage/cosmos/Cosmos.java [hunk 3]
```diff
@@ -45,18 +46,20 @@ public class Cosmos extends AbstractDistributedStorage {
   private final OperationChecker operationChecker;
 
   @Inject
-  public Cosmos(CosmosConfig config) {
+  public Cosmos(DatabaseConfig databaseConfig) {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/cosmos/Cosmos.java:49]: Made all the storage implementations receive a `DatabaseConfig` object.

File: core/src/main/java/com/scalar/db/storage/cosmos/CosmosConfig.java [hunk 4]
```diff
@@ -3,46 +3,38 @@
...
-  }
-
-  public CosmosConfig(Properties properties) {
-    super(properties);
-  }
-
-  @Override
-  protected void load() {
-    String storage = getProperties().getProperty(DatabaseConfig.STORAGE);
+  public CosmosConfig(DatabaseConfig databaseConfig) {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/cosmos/CosmosConfig.java:20]: Made constructors of `XXXConfig` receive a `DatabaseConfig` as well.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**